server/models/services/CollectionJobService.js

```diff
@@ -725,6 +732,119 @@ const updateCollectionJobTestStatusByQuery = ({
+ * - The Test Plan Report is the most recently "finalized" Report for the Test Plan Version
+ * - The most recent AT version used in any Test Plan Run is older than the specified AT Version
+ *
+ * @param {object} options
+ * @param {number} options.atVersionId - ID of the current AT version
+ * @param {*} options.transaction - Sequelize transaction
+ * @returns {Promise<Array>} - Array of created collection jobs
+ */
+const createCollectionJobsFromPreviousAtVersion = async ({
+  atVersionId,
+  transaction
+}) => {
+  const { currentVersion, previousVersionGroups } =
+    await getRerunnableTestPlanReportsForVersion({
+      currentAtVersionId: atVersionId,
+      transaction
+    });
+
+  if (!currentVersion || !previousVersionGroups) {
+    throw new Error('Failed to get rerunnable test plan reports');
+  }
+
+  if (previousVersionGroups.length === 0) {
+    return {
+      success: false,
+      description: `No rerunnable reports found for AT version ${currentVersion.name}`,
+      message: `No rerunnable reports found for AT version ${currentVersion.name}`
+    };
+  }
+
+  const collectionJobs = [];
+
+  for (const { reports } of previousVersionGroups) {
+    await Promise.all(
+      reports.map(async reportInfo => {
+        try {
+          const report = await TestPlanReport.findOne({
+            where: { id: reportInfo.id },
+            include: [
+              {
+                association: 'testPlanVersion',
+                include: [{ association: 'testPlan', required: true }],
+                required: true
+              },
+              { association: 'at', required: true }
+            ],
+            transaction
+          });
+
+          if (!report) return;
+
+          const newReport = await cloneTestPlanReportWithNewAtVersion(
+            report,
+            currentVersion,
+            transaction
+          );
+
+          const atVersion = await getAtVersionWithRequirements(
+            newReport.at.id,
+            currentVersion,
+            null,
+            transaction
+          );
+
+          if (!atVersion) return;
+
+          const job = await scheduleCollectionJob(
+            { testPlanReportId: newReport.id, atVersion },
+            { transaction }
+          );
+
+          collectionJobs.push(job);
+        } catch (error) {
+          console.error(
+            `Failed to create collection job for report ${reportInfo.id}:`,
+            error.message
+          );
+
+          await createUpdateEvent({
+            values: {
+              description: `Failed to start automated re-run for ${reportInfo.testPlanVersion.title} ${reportInfo.testPlanVersion.versionString} using ${currentVersion.at.name} ${currentVersion.name}: ${error.message}`,
+              type: UPDATE_EVENT_TYPE.COLLECTION_JOB
+            },
+            transaction
+          });
+        }
+      })
+    );
+  }
+
+  const message = `Created ${collectionJobs.length} re-run collection job${
```

Comment:
A disconnect I'm having around this feature is after hitting the 'Start Updates' button, I'm unsure on what has actually gotten re-ran by this, other than if I noted the items before or if I quickly view which bots have been triggered on the Test Queue.

Curious what others think when reviewed in use but perhaps it could be as simple as having a more detailed message outlining the the re-ran test plans.

Reply:
Yeah this makes sense to me. It would be trivial to list test plan version info in the update events but I was reluctant to do so out of fear of verbosity.